Tests/JWTKitTests/JWTKitTests.swift

```diff
@@ -94,34 +95,33 @@ class JWTKitTests: XCTestCase {
         }
     }
 
-    func testExpirationDecoding() throws {
+    func testExpirationDecoding() async throws {
         let data = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjIwMDAwMDAwMDB9.JgCO_GqUQnbS0z2hCxJLE9Tpt5SMoZObHBxzGBWuTYQ"
 
-        let test = try JWTSigner.hs256(key: "secret".bytes)
+        let test = try await JWTKeyCollection().addHS256(key: "secret".bytes)
             .verify(data, as: ExpirationPayload.self)
         XCTAssertEqual(test.exp.value, Date(timeIntervalSince1970: 2_000_000_000))
     }
 
-    func testExpirationEncoding() throws {
+    func testExpirationEncoding() async throws {
         let exp = ExpirationClaim(value: Date(timeIntervalSince1970: 2_000_000_000))
-        let jwt = try JWTSigner.hs256(key: "secret".bytes)
-            .sign(ExpirationPayload(exp: exp))
+        let keyCollection = await JWTKeyCollection().addHS256(key: "secret".bytes)
+        let jwt = try await keyCollection.sign(ExpirationPayload(exp: exp))
         let parser = try JWTParser(token: jwt.bytes)
         try XCTAssertEqual(parser.header(jsonDecoder: .defaultForJWT).typ, "JWT")
         try XCTAssertEqual(parser.header(jsonDecoder: .defaultForJWT).alg, "HS256")
         try XCTAssertEqual(parser.payload(as: ExpirationPayload.self, jsonDecoder: .defaultForJWT).exp, exp)
-        try parser.verify(using: .hs256(key: "secret".bytes))
+        try await parser.verify(using: keyCollection.signer()!.algorithm)
```

Comment:
One comment now I see the API, do we want `signer()` to throw instead of returning an optional? Having the optional here feels unnatural

Reply:
There's actually `require(kid:, alg:)` which I added later and wraps the `signer()` into a throwing function. Do you think we can use that or should we just make `signer` throwing and remove `require`?

Reply:
Just to be clear these aren't public anyways

Reply:
Do you need to test the internals then? Is it possible to rewrite the tests so that we still test the same behaviour but only via the public API (it should be). 

IMO a good test suite combined with a solid API doesn't need to use `@testable`

Reply:
I think I solved this specific instance but there's still some stuff we need to test which is not be public, like the ECDSA parameters, the `JWTParser` or the `PrimeGenerator`. Do you think we can make this any better?

Reply:
We can add those to a test file dedicated to internal tests. Feel free to create an issue and do it in a separate PR to unblock this